### PR TITLE
chapter 10 : Manage User

### PR DESCRIPTION
**Pull request by DaudKhadijah01**

### CODE REVIEW

**Comment by DaudKhadijah01** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,30 +1,82 @@
...
-    redirect_to root_path
+    if @user.nil?
+      flash[:danger] = t(:user_not_found)
+      redirect_to home_path
+    end
   end
 
   def new
-    @user = User.new
+     @user = User.new
```
Comment (ghost) [inline app/controllers/users_controller.rb:20]: remove extend space

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,30 +1,82 @@
...
-       redirect_to @user
+      log_in @user
+      flash[:success] = t(:welcome_greet)
+      redirect_to @user
     else
       render :new
     end
   end
 
+    def edit
```
Comment (ghost) [inline app/controllers/users_controller.rb]: align def with end at line 35

File: app/models/user.rb [hunk 3]
```diff
@@ -4,11 +4,11 @@ class User < ApplicationRecord
   validates :name, presence: true
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
   validates :email, presence: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}
-  validates :password, presence: true, length: {minimum: 6}
+  validates :password, presence: true, length: {minimum: 6}, allow_nil: true
 
   has_secure_password
 
-  class << self
+    class << self
```
Comment (ghost) [inline app/models/user.rb:12]: align with end at line 24

File: app/views/layouts/_header.html.erb [hunk 4]
```diff
@@ -10,21 +10,22 @@ inverse">
...
                 <%= link_to "#", class: "dropdown-toggle", "data-toggle": "dropdown" do %>
-                  <%= t("account") %>
+                  <%= t(:account) %>
                   <b class="caret"></b>
                 <% end %>
                 <ul class="dropdown-menu">
-                  <li><%= link_to t("profile"), current_user %></li>
-                  <li><%= link_to t("setting"), "#" %></li>
+                <li><%= link_to t(:profile), current_user %></li>
+                <li><%= link_to t(:setting), edit_user_path(current_user) %></li>
```
Comment (ghost) [inline app/views/layouts/_header.html.erb]: indent please

File: app/views/sessions/new.html.erb [hunk 5]
```diff
@@ -11,9 +11,9 @@
       <%= f.password_field :password, class: "form-control" %>
 
       <%= f.label :remember_me, class: "checkbox inline" do %>
-        <%= f.check_box :remember_me %>
-        <span>Remember me on this computer</span>
-      <% end %>
+          <%= f.check_box :remember_me %>
+          <span>Remember me on this computer</span>
+        <% end %>
```
Comment (ghost) [inline app/views/sessions/new.html.erb]: wrong indent

**Review by ghost — COMMENTED** [review]
cmt

**Comment by DaudKhadijah01** [comment]
ready

File: app/controllers/sessions_controller.rb [hunk 6]
```diff
@@ -2,19 +2,19 @@ class SessionsController < ApplicationController
   def new; end
 
   def create
-     @user = User.find_by email: params[:session][:email].downcase
-    if @user&.authenticate params[:session][:password]
+     @user = User.find_by email: params[:sessions][:email].downcase
```
Comment (huongnt-2545) [inline app/controllers/sessions_controller.rb:5]: why you change `sessions` instead of `session`??

**Review by huongnt-2545 — COMMENTED** [review]
Please resolve all comments of trainer

File: app/controllers/sessions_controller.rb (hunk 6 above)
Comment (DaudKhadijah01) [reply app/controllers/sessions_controller.rb:5]: because I got error if i use session

**Comment by DaudKhadijah01** [comment]
> Please resolve all comments of trainer

I already resolved it. Does my github not show any changes

**Comment by DaudKhadijah01** [comment]
ready

File: app/models/user.rb (hunk 3 above)
Comment (ngocvt-0484) [reply app/models/user.rb:12]: keep the same before

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**